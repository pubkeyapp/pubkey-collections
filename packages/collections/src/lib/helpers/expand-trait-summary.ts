import { CollectionTraitMap } from '../types/collection-trait-map'
import { CollectionTraitSummary } from '../types/collection-trait-summary'

export function expandTraitSummary(traits: CollectionTraitSummary): CollectionTraitMap {
  return Object.keys(traits).reduce((acc, key) => {
    acc[key] = traits[key].map((value) => ({
      key,
      value,
    }))
    return acc
  }, {} as CollectionTraitMap)
}
