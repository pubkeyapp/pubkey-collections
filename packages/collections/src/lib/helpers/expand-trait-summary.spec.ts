import { expandTraitSummary } from './expand-trait-summary'

describe('expand-trait-summary', () => {
  it('should expand traits', () => {
    // Arrange
    const traits = {
      Background: ['Red', 'Blue'],
      Eyes: ['Green'],
      Hair: [],
    }

    // Act
    const expanded = expandTraitSummary(traits)

    // Assert
    expect(expanded).toMatchSnapshot()
  })
})
