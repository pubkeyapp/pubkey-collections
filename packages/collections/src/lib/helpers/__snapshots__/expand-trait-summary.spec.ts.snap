// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expand-trait-summary should expand traits 1`] = `
{
  "Background": [
    {
      "key": "Background",
      "value": "Red",
    },
    {
      "key": "Background",
      "value": "Blue",
    },
  ],
  "Eyes": [
    {
      "key": "Eyes",
      "value": "Green",
    },
  ],
  "Hair": [],
}
`;
